### PR TITLE
Remove Newton's method from 'Easing.bezier()'

### LINKED ISSUE #6207: [Animation] Easing.bezier returning faulty values
This [CodePen](http://codepen.io/aleclarson/pen/VawJyJ?editors=0012) demonstrates the issue.

### PR DESCRIPTION
**Pull request by aleclarson**
Fixes #6207

### CODE REVIEW

File: Libraries/Animated/src/bezier.js [hunk 1]
```diff
@@ -49,14 +49,15 @@ module.exports = function(x1, y1, x2, y2, epsilon){
...
 
-    // First try a few iterations of Newton's method -- normally very fast.
-    for (t2 = x, i = 0; i < 8; i++){
-      x2 = curveX(t2) - x;
-      if (Math.abs(x2) < epsilon) { return curveY(t2); }
-      d2 = derivativeCurveX(t2);
-      if (Math.abs(d2) < 1e-6) { break; }
-      t2 = t2 - x2 / d2;
-    }
+    // TODO: Figure out why this produces faulty values.
```
Comment (vjeux) [inline Libraries/Animated/src/bezier.js]: Can you remove it altogether?

**Comment by vjeux** [comment]
Thanks! Let's just remove the code instead of commenting it

**Comment by aleclarson** [comment]
@vjeux I knew you would say that! Do you not care to see Newton's method fixed? :stuck_out_tongue:

**Comment by vjeux** [comment]
If you fix it, I'll take it. But I don't want to have the code littered with commented out sections that may or may not work. In my experience it doesn't get things fixed, instead it makes people not want to touch this part of the codebase in fear or breaking something.

One thing you can do is to add a comment that says that it is possible to make it faster by implementing a few rounds of newton. It will trigger people's competitive aspects and get them to implement it :)

**Comment by aleclarson** [comment]
@gre has a [bezier-easing](https://github.com/gre/bezier-easing/blob/master/index.js) repo that implements Newton's method differently. Perhaps he can help! :smile: 

But for now, I'll just remove the commented code so we can get this merged. :wink:

**Comment by vjeux** [comment]
@facebook-github-bot shipit

**Comment by gre** [comment]
@aleclarson yeah, `bezier-curve` is basically inspired from Firefox and Chrome implementation. It uses different strategies to have a precise and efficient interpolation.

> - A first optimization store sample values of the bezier curve in a small table used to roughly find a initial X guess.
- Then, it use two different implementation strategies: One use the Newton’s method and the other is just a dichotomic search (binary subdivision).
- A criteria based on the slope give the best strategy to take.

http://greweb.me/2012/02/bezier-curve-based-easing-functions-from-concept-to-implementation/

it is also used here: http://greweb.me/bezier-easing-editor/example/

**Comment by aleclarson** [comment]
@gre Sweet! That's a perfect guide if anyone decides to make `Easing.bezier()` more performant.

@vjeux It looks like Travis CI build is failing. I don't think it's my fault though.

**Comment by vjeux** [comment]
@gre if you are interested in sending a pull request to use your implementation (that I trust much more), I'd be happy to pull it in.

@aleclarson do you mind rebasing on-top of master?

**Comment by gre** [comment]
@vjeux Yeah I can send a PR. Do you prefer I embed it in react-native or depend on the npm lib?

**Comment by vjeux** [comment]
@facebook-github-bot shipit

**Comment by aleclarson** [comment]
Is this PR needed anymore then? The test might be worth merging, but maybe @gre can come up with a better one.

**Comment by gre** [comment]
maybe it can still get in and we improve it after that?

**Comment by vjeux** [comment]
If you can embed it and remove all the css, toString, generic parts... that are not useful for this only call site that would be awesome. We're trying to keep the core as small as possible as it directly impacts startup time.

https://github.com/facebook/react-native/blob/master/Libraries/Animated/src/Easing.js#L117

**Comment by mkonicek** [comment]
Internal CI flakiness?

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by aleclarson** [comment]
Closing this in favor of #6433 👏

**Pull request closed without merging**